lib/azure/armrest/configuration.rb

```diff
@@ -93,8 +96,12 @@ def initialize(args)
           :grant_type   => 'client_credentials',
           :proxy        => ENV['http_proxy'],
           :ssl_version  => 'TLSv1',
+          :max_threads  => 10
         }.merge(args.symbolize_keys)
 
+        # Avoid thread safety issues for VCR testing.
+        options[:max_threads] = 1 if defined?(VCR)
+
```

Comment:
We don't. If you don't like this approach I can remove it, but it does mean some stubbing in our refresher specs.